### ISSUE TITLE
It generates HTML panels instead of charts for untyped metrics

### ISSUE THREAD

**Issue by Lusitaniae** [FUSAKLA/autograf#2]
Given this input

```
cat /tmp/prom 
# HELP quic_connections_stream_read_timeouts InfluxDB Metric
# TYPE quic_connections_stream_read_timeouts untyped
quic_connections_stream_read_timeouts{host_id="3c2G4LXc9qnWZBSYCs53fC7hy9SVJbEquofzCEULMtpr"} 0
```

It generates HTML panel with the metric name

```
autograf --metrics-file /tmp/prom  | jq
{
  "slug": "",
  "title": "Autograf dashboard",
  "originalTitle": "",
  "tags": [
    "autograf",
    "generated"
  ],
  "style": "dark",
  "timezone": "browser",
  "editable": true,
  "hideControls": false,
  "sharedCrosshair": true,
  "panels": null,
  "rows": [
    {
      "title": "quic",
      "showTitle": true,
      "collapse": true,
      "editable": true,
      "height": "250px",
      "panels": [
        {
          "editable": false,
          "error": false,
          "gridPos": {},
          "id": 1,
          "isNew": false,
          "renderer": "flot",
          "span": 6,
          "title": "Panel Title",
          "transparent": false,
          "type": "text",
          "content": "unknown type unknown of metric quic_connections_stream_read_timeouts",
          "mode": "html",
          "pageSize": 0,
          "scroll": false,
          "showHeader": false,
          "sort": {
            "col": 0,
            "desc": false
          },
          "styles": null,
          "fieldConfig": {
            "defaults": {
              "unit": "",
              "color": {
                "mode": ""
              },
              "thresholds": {
                "mode": "",
                "steps": null
              },
              "custom": {
                "axisPlacement": "",
                "barAlignment": 0,
                "drawStyle": "",
                "fillOpacity": 0,
                "gradientMode": "",
                "lineInterpolation": "",
                "lineWidth": 0,
                "pointSize": 0,
                "showPoints": "",
                "spanNulls": false,
                "hideFrom": {
                  "legend": false,
                  "tooltip": false,
                  "viz": false
                },
                "lineStyle": {
                  "fill": ""
                },
                "scaleDistribution": {
                  "type": ""
                },
                "stacking": {
                  "group": "",
                  "mode": ""
                },
                "thresholdsStyle": {
                  "mode": ""
                }
              }
            },
            "overrides": null
          },
          "options": {
            "content": "",
            "mode": ""
          }
        }
      ],
      "repeat": null
    }
  ],
  "templating": {
    "list": [
      {
        "name": "datasource",
        "type": "datasource",
        "datasource": null,
        "refresh": 1,
        "options": [],
        "includeAll": false,
        "allFormat": "",
        "allValue": "",
        "multi": false,
        "multiFormat": "",
        "query": "prometheus",
        "regex": "//",
        "current": {
          "text": null,
          "value": null
        },
        "label": "datasource",
        "hide": 0,
        "sort": 0
      },
      {
        "name": "job",
        "type": "query",
        "datasource": {
          "type": "",
          "UID": ""
        },
        "refresh": 1,
        "options": [
          {
            "text": "All",
            "value": "$__all",
            "selected": false
          }
        ],
        "includeAll": true,
        "allFormat": "",
        "allValue": ".*",
        "multi": true,
        "multiFormat": "",
        "query": "query_result()",
        "regex": "/job=\"([^\"]+)\"/",
        "current": {
          "text": [
            "All"
          ],
          "value": "$__all"
        },
        "label": "job",
        "hide": 0,
        "sort": 1
      },
      {
        "name": "instance",
        "type": "query",
        "datasource": {
          "type": "",
          "UID": ""
        },
        "refresh": 1,
        "options": [
          {
            "text": "All",
            "value": "$__all",
            "selected": false
          }
        ],
        "includeAll": true,
        "allFormat": "",
        "allValue": ".*",
        "multi": true,
        "multiFormat": "",
        "query": "query_result()",
        "regex": "/instance=\"([^\"]+)\"/",
        "current": {
          "text": [
            "All"
          ],
          "value": "$__all"
        },
        "label": "instance",
        "hide": 0,
        "sort": 1
      }
    ]
  },
  "annotations": {
    "list": null
  },
  "refresh": "1m",
  "schemaVersion": 0,
  "version": 0,
  "links": null,
  "time": {
    "from": "now-1h",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ],
    "time_options": [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d"
    ]
  }
}

```

If the metric description is confusing (the InflxuDB stuff), looks like removing the comments doesn't help either as it creates an empty dashboard (no charts)

```
cat /tmp/prom 
quic_connections_stream_read_timeouts{host_id="3c2G4LXc9qnWZBSYCs53fC7hy9SVJbEquofzCEULMtpr"} 0
```

```
{
  "slug": "",
  "title": "Autograf dashboard",
  "originalTitle": "",
  "tags": [
    "autograf",
    "generated"
  ],
  "style": "dark",
  "timezone": "browser",
  "editable": true,
  "hideControls": false,
  "sharedCrosshair": true,
  "panels": null,
  "rows": [],
  "templating": {
    "list": [
      {
        "name": "datasource",
        "type": "datasource",
        "datasource": null,
        "refresh": 1,
        "options": [],
        "includeAll": false,
        "allFormat": "",
        "allValue": "",
        "multi": false,
        "multiFormat": "",
        "query": "prometheus",
        "regex": "//",
        "current": {
          "text": null,
          "value": null
        },
        "label": "datasource",
        "hide": 0,
        "sort": 0
      },
      {
        "name": "job",
        "type": "query",
        "datasource": {
          "type": "",
          "UID": ""
        },
        "refresh": 1,
        "options": [
          {
            "text": "All",
            "value": "$__all",
            "selected": false
          }
        ],
        "includeAll": true,
        "allFormat": "",
        "allValue": ".*",
        "multi": true,
        "multiFormat": "",
        "query": "query_result()",
        "regex": "/job=\"([^\"]+)\"/",
        "current": {
          "text": [
            "All"
          ],
          "value": "$__all"
        },
        "label": "job",
        "hide": 0,
        "sort": 1
      },
      {
        "name": "instance",
        "type": "query",
        "datasource": {
          "type": "",
          "UID": ""
        },
        "refresh": 1,
        "options": [
          {
            "text": "All",
            "value": "$__all",
            "selected": false
          }
        ],
        "includeAll": true,
        "allFormat": "",
        "allValue": ".*",
        "multi": true,
        "multiFormat": "",
        "query": "query_result()",
        "regex": "/instance=\"([^\"]+)\"/",
        "current": {
          "text": [
            "All"
          ],
          "value": "$__all"
        },
        "label": "instance",
        "hide": 0,
        "sort": 1
      }
    ]
  },
  "annotations": {
    "list": null
  },
  "refresh": "1m",
  "schemaVersion": 0,
  "version": 0,
  "links": null,
  "time": {
    "from": "now-1h",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ],
    "time_options": [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d"
    ]
  }
}

```

**Comment by Lusitaniae** [comment]
After replacing untyped with counter, that seems to generate dashboards


Something else I'm seeing is that they're getting generated with errors:

`,job` -> `{job`

```
              "expr": "rate(quic_client_connection_stats_num_packets,job=~'$job',instance=~'$instance'}[3m])",
```

**Comment by FUSAKLA** [comment]
Hi, thanks for rising the issue!

The first problem regarding unknown metric type, was actually not an issue.
I just decided that if the metric type is not any of the expected, autograf creates just text panel with warning that it's unknown.

The second issue is valid and caused if the `--selector` flag is not set which is your case. 

I created a fix for both, see https://github.com/FUSAKLA/autograf/pull/3
